pkg/acceptance: update gss tests with container built from master

spilchen opened the pull request:
I previously updated the GSS acceptance tests using a container built from my fork. Now that the related server changes are in master, I’m updating the container tags to use an image built from master for reproducibility.

This also will refresh the container with one that is built for both arm64 and amd64. This should address #132232.

See this for more context: https://cockroachlabs.slack.com/archives/CJ0H8Q97C/p1728061862175869

Epic: CRDB-39988
Informs CRDB-41998
Closes #132232 
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132249)
<!-- Reviewable:end -->

spilchen committed:
pkg/acceptance: update gss tests with container built from master

I previously updated the GSS acceptance tests using a container built
from my fork. Now that the related server changes are in master, I’m
updating the container tags to use an image built from master for
reproducibility.

Epic: CRDB-39988
Informs CRDB-41998
Closes #132232
Release note: none

spilchen commented:
TFTR!

bors r=rail
